File: backend/catalog-service/src/main/java/me/doruk/catalogservice/request/VenueCreateRequest.java
package me.doruk.catalogservice.request;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import jakarta.validation.constraints.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class VenueCreateRequest {
  @NotBlank
  @Size(max = 255)
  private String name;

  @NotBlank
  @Size(max = 255)
  private String location;

  @Min(1)
  private int totalCapacity;

  @Size(max = 512)
  private String imageUrl;
}

File: backend/catalog-service/Requests/VenueCreateRequest.cs
using System.ComponentModel.DataAnnotations;

namespace CatalogService.Requests;

public class VenueCreateRequest
{
    [Required]
    [StringLength(255)]
    public string Name { get; set; } = default!;

    [Required]
    [StringLength(255)]
    public string Location { get; set; } = default!;

    [Range(1, int.MaxValue)]
    public int TotalCapacity { get; set; }

    [StringLength(512)]
    public string? ImageUrl { get; set; }
}
